packages/search-field/src/components/ScrollBox.js

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import PropTypes from "prop-types";
+import debounce from "lodash.debounce";
+
+import ScrollWrapper from "../styledComponents/ScrollWrapper";
+
+export default class ScrollBox extends React.Component {
+  state = {
+    atEnd: false
+  };
+
+  componentDidMount = () => {
+    this.wrapper && this.wrapper.addEventListener("scroll", this.onScroll);
+  };
+  componentWillUnmount = () => {
+    this.wrapper && this.wrapper.removeEventListener("scroll", this.onScroll);
+  };
+
+  centerChildByIndex = index => {
+    const element = this.wrapper.childNodes[index];
+    if (!element) return;
+    const elementCenter = element.offsetTop + element.clientHeight / 2;
+    this.wrapper.scrollTop =
+      elementCenter - this.wrapper.offsetTop - this.wrapper.clientHeight / 2;
+  };
+
+  checkScrollPosition = () => {
+    const reachEnd =
+      this.wrapper.scrollHeight -
+        this.wrapper.scrollTop -
+        this.wrapper.clientHeight <
+      this.wrapper.clientHeight / 2;
```

Comment:
maybe a comment about this math? or break the parts on meaningful variable names like `remainingContent < halfVisibleHeight`I dont know...